### ISSUE TITLE
Build fails with Boost < v1.58

### ISSUE THREAD

**Issue by Fuzzbawls** [PIVX-Project/PIVX#65]
```
libbitcoin_server.a(libbitcoin_server_a-init.o): In function `copy_file':
/usr/include/boost/filesystem/operations.hpp:384: undefined reference to `boost::filesystem::detail::copy_file(boost::filesystem::path const&, boost::filesystem::path const&, boost::filesystem::copy_option, boost::system::error_code*)'
libbitcoin_wallet.a(libbitcoin_wallet_a-walletdb.o): In function `copy_file':
/usr/include/boost/filesystem/operations.hpp:381: undefined reference to `boost::filesystem::detail::copy_file(boost::filesystem::path const&, boost::filesystem::path const&, boost::filesystem::copy_option, boost::system::error_code*)'
```

Potential fix in #64

**Comment by presstab** [comment]
I actually do not have any verification that 1.55 is the cutoff. I know less than 1.54 would not compile. I know 1.58 does compile. I thought someone said they built fine with 1.55, but not completely sure.

**Comment by jonspock** [comment]
I can track it down tomorrow. Sorry this broke. I think I didn't notice because it was OK with the script I used.

**Comment by Fuzzbawls** [comment]
I don't have a setup personally that uses 1.55, though its simple enough to get going on ubuntu 14.04 by doing `sudo apt-get install libboost1.55-all-dev`

depends and gitian both use Boost 1.59 and don't have this issue, so at least we know this won't affect the release process

**Comment by Fuzzbawls** [comment]
Tested this with Boost v1.55, and indeed it also produces the error. Since we have 3rd party conformation that the error does NOT exist with Boost 1.58, lets just go ahead and use that as the baseline version.

**Comment by jonspock** [comment]
I looked into this and it seems the error is only because boost was built without the C++11 option while you are linking after building with C++11.  So a system that uses it's own boost such as gitian or s3v3nh4cks script will be OK because it is consistent, while using the default boost in cases will have problems. In our case this is specifically older Ubuntu distributions. So a better fix may be to disable C++11 on older Linux distributions since it was mainly needed for QT 5.8 and above which wouldn't exist on the older Linux distros.

**Comment by Fuzzbawls** [comment]
I'm actually OK with the current fix as-is. While upstream dependencies on c++11 availability will surely reintroduce this error, it will also drive the point of not relying on distribution packages, ESPECIALLY with the migration to more recent versions of Qt.

At some point we need to say "enough with distro libs" and use the intended versions and build options, but for now, I think this is sane.

**Comment by Fuzzbawls** [comment]
followup: 3rd party services such as exchanges should already be aware of these types of issues due to having a dependency on bitcoin (unless they are reluctant to upgrade their bitcoin version) and should already be aware of the depends system, which negates this error completely.

**Comment by presstab** [comment]
I think the case as is, should be fine. Backing up the wallet is done once per session, and a slightly less optimal way of backing up the file shouldn't be too big of a deal.

If I different fix is chosen, whatever it is it should be automatically done so that the user doesn't have to search out the issue and figure out special configure flags to use.

**Comment by Fuzzbawls** [comment]
Fixed in #66, closing.
